handler/src/test/java/io/netty/handler/ssl/SslHandlerTest.java

```diff
@@ -431,6 +437,97 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
         assertTrue(events.isEmpty());
     }
 
+    @Test(timeout = 5000)
+    public void testHandshakeFailBeforeWritePromise() throws Exception {
+        SelfSignedCertificate ssc = new SelfSignedCertificate();
+        final SslContext sslServerCtx = SslContextBuilder.forServer(ssc.certificate(), ssc.privateKey()).build();
+        final CountDownLatch latch = new CountDownLatch(2);
+        final CountDownLatch latch2 = new CountDownLatch(2);
+        final BlockingQueue<Object> events = new LinkedBlockingQueue<Object>();
+        Channel serverChannel = null;
+        Channel clientChannel = null;
+        EventLoopGroup group = new DefaultEventLoopGroup();
+        try {
+            ServerBootstrap sb = new ServerBootstrap();
+            sb.group(group)
+                .channel(LocalServerChannel.class)
+                .childHandler(new ChannelInitializer<Channel>() {
+                  @Override
+                  protected void initChannel(Channel ch) {
+                      ch.pipeline().addLast(sslServerCtx.newHandler(ch.alloc()));
+                      ch.pipeline().addLast(new ChannelInboundHandlerAdapter() {
+                          @Override
+                          public void channelActive(ChannelHandlerContext ctx) {
+                              ByteBuf buf = ctx.alloc().buffer(10);
+                              buf.writerIndex(buf.readerIndex() + 10);
```

Comment:
nit: you could also just call `buf.writeZero(10)`.

Reply:
done